File: design-patterns/ts/strategy/calculator/calculation-strategy.ts
// calculation-strategy.ts

import Calculation from "./calculation";

class CalculationStrategy {
    private calculation: Calculation;

    constructor(calculation: Calculation) {
        this.calculation = calculation;
    }

    execute(num1: number, num2: number): number {
        return this.calculation.execute(num1, num2);
    }
}

export default CalculationStrategy;

File: design-patterns/go/strategy/math_calculation/calculation_strategy.go
// calculation_strategy.go

package main

type CalculationStrategy struct {
	calculation Calculation
}

func NewCalculationStrategy(calculation Calculation) (calculationStrategy *CalculationStrategy) {
	calculationStrategy = &CalculationStrategy{}
	calculationStrategy.calculation = calculation
	return
}

func (calculationStrategy *CalculationStrategy) Execute(num1 float32, num2 float32) (float32) {
	return calculationStrategy.calculation.Execute(num1, num2)
}